ls-ecaps.c

```diff
@@ -690,35 +690,49 @@ cap_rcec(struct device *d, int where)
 }
 
 static void
-cap_dvsec_cxl(struct device *d, int id, int where)
+dvsec_cxl_device(uint8_t *data, int rev)
 {
   u16 w;
 
-  printf(": CXL\n");
-  if (verbose < 2)
-    return;
-
-  if (id != 0)
-    return;
-
-  if (!config_fetch(d, where + PCI_CXL_CAP, 0x38 - 0xa))
+  /* Legacy 1.1 revs aren't handled */
+  if (rev != 1)
```

Comment:
Cannot be revision bigger than 1 in future with backward compatible capabilities? IIRC All PCI capabilities were always defined in backward compatible manner. So check for one specific revision seems to be unexpected.

Reply:
Yes, you are correct. CXL capabilities are defined to be backwards compatible with future revisions. Checking against a specific revision of capabilities without also checking future revisions prevents forward compatibility with later revisions. I'll push an amended commit addressing this.